src/interactivity/win32/windowio.cpp

```diff
@@ -78,7 +78,7 @@ VOID SetConsoleWindowOwner(const HWND hwnd, _Inout_opt_ ConsoleProcessHandle* pP
     else
     {
         // Find a process to own the console window. If there are none then let's use conhost's.
-        pProcessData = gci.ProcessHandleList.GetFirstProcess();
+        pProcessData = gci.ProcessHandleList.GetOldestProcess();

```

Comment:
This is a behavior change: `GetFirstProcess()` used to return the newest process, but I think this was wrong... conhost's PID was the last (= oldest) entry if anything. So I've chosen to rename the function to better convey its meaning.

Reply:
> [any old](https://en.wiktionary.org/wiki/any_old) has an idiomatic meaning, though.

Ah good point.

> Is the comment "_If there are none then let's use conhost's._" still accurate, or does this now use conhost's process even if others exist?

I'm not sure it ever used to be accurate. `conhost` itself isn't part of the process lists - console clients are (i.e. shells and other console subsystem applications). I wouldn't know how conhost could be its own client.
In reality this code path is only invoked via [`Window::SetOwner`](https://github.com/microsoft/terminal/blob/437b5ac595915830194498eca8825e6037e288c0/src/host/srvinit.cpp#L282) when the "root process" exits (the first process that is launched - usually your shell, like `cmd.exe` for instance), but there are still other clients attached. This is for instance reproducible if you run `pwsh.exe` inside `cmd.exe` and close the window. Both processes will get get a close event simultaneously, but PowerShell is slow and so it'll take longer to exit then `cmd.exe`. Once `cmd.exe` is gone, `conhost` will need a new "root process" and in this case that would be `pwsh.exe` (until it also exits a hundred or so milliseconds later).

So basically, the comment is at least generally wrong/inaccurate, but I don't want to touch it, because it might prove to be a useful nugget of information if looked up via `git blame` in the future. The old code used to use the newest process as the new root process and that seems like a mistake to me. If anything it should be the oldest one as far as I can see.